File: backend-locadora/src/main/java/com/locadora/backendlocadora/model/Titulo.java
package com.locadora.backendlocadora.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "titulos")
public class Titulo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nome;

    @Column(nullable = false)
    private String ano;

    @Column(nullable = false)
    private String sinopse;

    @Column(nullable = false)
    private String categoria;

    @ManyToOne
    @JoinColumn(name = "id_diretor", referencedColumnName = "id")
    private Diretor diretor;

    @ManyToOne
    @JoinColumn(name = "id_classe", referencedColumnName = "id")
    private Classe classe;

    @ManyToMany
    @JoinTable(name = "titulo_ator",
            joinColumns = @JoinColumn(name = "id_titulo"),
            inverseJoinColumns = @JoinColumn(name = "id_ator"))
    private List<Ator> atores;

}

File: frontend-locadora/src/app/modulos/titulos/model/titulo.ts
import { BaseModel } from "src/app/shared/base/base.model";
import { Ator } from "../../atores/model/ator";
import { Classe } from "../../classes/model/classe";
import { Diretor } from "../../diretores/model/diretor";

export interface Titulo extends BaseModel {
  nome: string;
  ano: string;
  sinopse: string;
  categoria: string;
  classe: Classe;
  diretor: Diretor;
  atores: Ator[]
}
